### PR TITLE
Update compiler.md

### PR DESCRIPTION
**Pull request by aziyatali**
The sentence is very confusing. Excessive/invalid usage of English grammar

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [x] Errors have a helpful link attached, see [`contributing.md`](https://github.com/vercel/next.js/blob/canary/contributing.md)

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] [e2e](https://github.com/vercel/next.js/blob/canary/contributing/core/testing.md#writing-tests-for-nextjs) tests added
- [x] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have a helpful link attached, see [`contributing.md`](https://github.com/vercel/next.js/blob/canary/contributing.md)

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm build && pnpm lint`
- [ ] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

### CODE REVIEW

File: docs/advanced-features/compiler.md [hunk 1]
```diff
@@ -77,7 +77,7 @@ module.exports = {
 
 ### Jest
 
-Jest support not only includes the transformation previously provided by Babel, but also simplifies configuring Jest together with Next.js including:
+Jest not only includes the transformation previously provided by Babel, but also simplifies configuring Jest together with Next.js including:
```
Comment (styfle) [inline docs/advanced-features/compiler.md:80]:
```suggestion
The Next.js Compiler also performs transformation for your Jest tests and simplifies configuring Jest together with Next.js including:
```

**Review by styfle — APPROVED** [review]

**Comment by aziyatali** [comment]
@styfle  Steven, transpiles(aka: transpiler) sounds hard for beginner programmers. Would be better if we could find some alternative!

**Comment by styfle** [comment]
What do you suggest?

**Comment by aziyatali** [comment]
@styfle Let's leave as it is. If they won't understand then they will search for it. At worst case they will learn new concept `source-to-source' compiler!

**Pull request merged**